core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -665,14 +665,15 @@ private WorkerInfo extractWorkerInfo(MasterWorkerInfo worker,
   public void removeBlocks(List<Long> blockIds, boolean delete) throws UnavailableException {
     try (JournalContext journalContext = createJournalContext()) {
       for (long blockId : blockIds) {
-        HashSet<Long> workerIds = new HashSet<>();
-
+        Set<Long> workerIds;
         try (LockResource r = lockBlock(blockId)) {
           Optional<BlockMeta> block = mBlockStore.getBlock(blockId);
           if (!block.isPresent()) {
             continue;
           }
-          for (BlockLocation loc : mBlockStore.getLocations(blockId)) {
+          List<BlockLocation> locations = mBlockStore.getLocations(blockId);
+          workerIds = new HashSet<>(locations.size());
+          for (BlockLocation loc : locations) {
```

Comment:
@dbw9580 Do you happen to know how I can do things like below?
```
new HashSet<>(mBlockStore.getLocations(blockId).stream().map(BlockLocation::getWorkerId))
```
Basically I want to use the sized constructor. 
```
list.stream.map(...).collect(Collectors.toSet)
```
This is actually using the non-sized constructor of HashSet and repeatedly doing set.addAll() which causes resizing.

Reply:
`.collect(Collectors.toCollection(() -> new HashSet<>(locations.size())))`

The readability is subtly inferior to a for loop, IMO.

Reply:
And does not using streams create even more intermediate, ephemeral objects behind the scenes, considering this piece of code is inside an outer for loop?

Reply:
Ok i'll stick to for loop then. Thx :)